lib/call.ts

```diff
@@ -26,21 +27,11 @@ class CallTranspiler extends base.TranspilerStep {
         }
         this.visitCall(<ts.NewExpression>node);
         break;
-      case ts.SyntaxKind.ArrayLiteralExpression:
-        if (this.isInsideConstExpr(node)) {
-          this.emit('const');
-        }
-        return false;
-      case ts.SyntaxKind.ObjectLiteralExpression:
-        if (this.isInsideConstExpr(node)) {
-          this.emit('const');
-        }
-        return false;
       case ts.SyntaxKind.CallExpression:
         var callExpr = <ts.CallExpression>node;
-        if (!this.maybeHandleSuperCall(callExpr)) {
-          this.visitCall(callExpr);
-        }
+        if (this.fc.handleCall(callExpr)) break;
```

Comment:
Done re maybe convention.
Regarding the ||, I think this is not much more code and reads a bit cleaner to me.